spec/features/create_object_h2_spec.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+RSpec.describe 'Use H2 to create an object', type: :feature do
+  let(:collection_title) { RandomWord.nouns.next }
+  let(:item_title) { "SUL Logo for #{collection_title}" }
+  let(:start_url) { "#{Settings.h2_url}/dashboard" }
+  let(:user_email) { "#{AuthenticationHelpers.class_variable_get(:@@username)}@stanford.edu" }
+
+  before do
+    authenticate!(start_url: start_url,
+                  expected_text: 'Your collections')
+  end
+
+  scenario do
+    click_link '+ Create a new collection'
+
+    # Checks for specific content on create collection view
+    expect(page).to have_content('Manage release of deposits for discovery and download')
+
+    # Fills in basic collection information and saves
+    fill_in 'Collection name', with: collection_title
+    fill_in 'Description', with: "Integration tests for #{collection_title}"
+    fill_in 'Contact email', with: user_email
+
+    # Adds user to depositor field
+    fill_in 'Depositors', with: AuthenticationHelpers.class_variable_get(:@@username)
+
+    click_button 'Deposit'
+
+    expect(page).to have_content(collection_title)
+
+    # Deposit an item to the collection
+    find('header', text: collection_title).sibling('button').click
+
+    # Selects image type
+    find('label', text: 'Image').click
+
+    # Route to work deposit view
+    click_button 'Continue'
+
+    # Work Deposit view
+    expect(page).to have_content('Deposit your work')
+    attach_file('spec/fixtures/sul-logo.png') do
+      find('div.dropzone').click
+    end
+    expect(page).to have_content('sul-logo.png')
+    fill_in 'Title of deposit', with: item_title
+    fill_in 'Contact email', with: user_email
+    fill_in 'First name', with: 'Dana'
+    fill_in 'Last name', with: 'Scully'
+    fill_in 'Abstract', with: "An abstract for #{collection_title} logo"
+
+    fill_in 'Citation', with: 'All rights reserved'
+
+    find('input#work_agree_to_terms').click
+
+    find_button('Deposit').click
```

Comment:
This triggers an asynchronous action.  How does it wait until the deposit is finished?

Reply:
I believe the `expect` statement on line 60 waits until the deposit is finished.

Reply:
Line 60 is a synchronous reponse though.  That will be printed out right away while a background job will transfer the file to SDR.

Reply:
Added `reload_page_until_timeout` call that waits for the druid to appear on work status page.